### DISCUSSION TITLE
Error with predict()

### DISCUSSION THREAD

**Discussion by Prinjesh** [Lightning-AI/pytorch-lightning#7747] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am trying to predict on my test data but it throws an error:
TypeError: conv2d(): argument 'input' (position 1) must be Tensor, not list

To help you get context:
I have called a pre-trained model and am trying to predict on the model parameters. The model is resnet18().

The `test_loader` is a dataloader that loads images from a test image folder. The folder has labelled folders of the test files.
   
```py
# test dataloader
test_dataset = torchvision.datasets.ImageFolder(
    'path/to/test_data_labelled_folders',
    transform=torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
)

testloader = torch.utils.data.DataLoader(
    test_dataset, num_workers=4, batch_size=4)

# loading previously trained model 
model = LitClassifier.load_from_checkpoint('path/to/checkpoint.ckpt') #LitClassifier is a Lightningmodule

# calling predict function
trainer = pl.Trainer()
trainer.predict(model=model, dataloaders=[testloader], return_predictions=True)
```

**Comment by justusschock** [comment, marked as answer]
Did you overwrite the `predict_step`? By default it just feeds the whole batch through `forward` (which with the image folder also includes the label and therefore is a list)

So you have two choices: Remove the labels from you predict data or overwrite the predict step to ignore them :)
